python/set_dev_version.py

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Set development version from git
+
+This is used to set the Python package version before building
+nightly wheels, since at least some tools do not allow
+duplicate versions when uploading sdist/wheels. This requires
+that the repo has a full checkout with tags.
+
+The Python package version is hard-coded in a _static_version.py
```

Comment:
Alternatively, we could also use more of the original https://github.com/jbweston/miniver code (the version setup we now have in Python is a slimmed down version of that, to remove the "get version from git" part)